Feature/wg 109

Evil00787 opened the pull request:
Added Routes and Search logic

Evil00787 committed:
Added logic of Route loading and storing
TODO: Displaying UI, playing route after choosing.
Problem: How to get to NodeController::LoadNode function from RoutesLoader.

Evil00787 committed:
Resolved last problem.
Added displaying RoutesUI.
Much to improve.

stasgora committed:
Merge branch 'master' into feature/WG-109

# Conflicts:
#	Assets/WIKIGRAPH.unity

stasgora committed:
Merge

Evil00787 committed:
Resolved problem with RouteNames
Resolved problem with display Routes in RoutesUI

Evil00787 committed:
Fixed Routes architecture - moved routes section from GraphController to the HistoryController

Evil00787 committed:
Minor changes in HistoryController

stasgora committed:
Merged?

Evil00787 committed:
Merged with master

Merge branch 'master' into feature/WG-109

Evil00787 committed:
MERGED
Merge branch 'feature/WG-109' of https://github.com/wikitools/wikigraph into feature/WG-109

Evil00787 committed:
CHanged HistoryController

Evil00787 committed:
Merging
Merge branch 'master' into feature/WG-109

Evil00787 committed:
Merging...

Evil00787 committed:
Resolved bug with first node in history

Evil00787 committed:
Updated line in NodeController - too tight restriction on selecting Nodes, might be changed later

Evil00787 committed:
Deleted RoutesUI

Evil00787 committed:
Added starting routes functionality - now you can test in with the button
Resolved all bugs with reading files and playing routes

Evil00787 committed:
Added possibility to stop route playing after exit click

Evil00787 committed:
Merge branch 'master' into feature/WG-109

Evil00787 committed:
Merge ...

Evil00787 committed:
Scene improvement

Evil00787 committed:
Added displaying available routes

Evil00787 committed:
You can now start routes from console!
Added displaying length of route

Evil00787 committed:
Console behaviour on playing routes works properly - no more bugs after starting multiple routes

Evil00787 committed:
Some modifications

stasgora committed:
Merge?

stasgora committed:
Fixes & scene cleanup

Evil00787 committed:
First working version (hardcoded search file) of Search!
Have fun :)

Evil00787 committed:
Changed method of string comparison - culture

stasgora committed:
Reader path func

Evil00787 committed:
Routes depend on data pack
Added reading .wgs files caontaing titles of categories and articles!
Format:

(string);(uint)

Evil00787 committed:
Corrected search for .wgs files

Evil00787 committed:
Fixed search click

Evil00787 committed:
Added option to StopRoute from console
Any userActions stop playing route

Evil00787 committed:
Fixed stoping route on changeMode

Evil00787 committed:
All other actions now stop Route
Moving scroll to the top on searching

Evil00787 committed:
Fixed routes names

Evil00787 committed:
FIxed gettting routename

stasgora commented on Assets/Scripts/Controllers/HistoryController.cs:
```diff
@@ -21,24 +59,163 @@ public class HistoryController : MonoBehaviour {
+			uint index;
+			if (uint.TryParse(EventSystem.current.currentSelectedGameObject.name, out index)) {
+				nodeChangedSource = NodeChangedSource.Search;
+				Node node = nodeController.NodeLoadManager.LoadNode(index);
```
Right after this line a force sync is required, to load the node before it is to be selected on clients
```suggestion
				Node node = nodeController.NodeLoadManager.LoadNode(index);
				nodeController.OnNodeLoadSessionEnded?.Invoke();
```

stasgora replied:
Maybe extract those 3 lines to a private function

stasgora commented on Assets/Scripts/Controllers/HistoryController.cs:
```diff
@@ -21,24 +59,163 @@ public class HistoryController : MonoBehaviour {
+			Dictionary<uint, string> searchResults = HistoryService.searchLoader.getEntries(index);
+			int i = 0;
+			foreach (var result in searchResults) {
+				Node node = nodeController.NodeLoadManager.LoadNode(result.Key);
```
We shouldn't load all nodes just to check their type, `NodeLoader` now has:
`GetNodeType(id)`

stasgora commented on Assets/Scripts/Controllers/HistoryController.cs:
```diff
@@ -21,24 +59,163 @@ public class HistoryController : MonoBehaviour {
-					nodeChangedByHistory = true;
+				NodeSelectedAction.selectNodeAction = (node, isRoute) => {
+					if (isRoute) nodeChangedSource = NodeChangedSource.Route;
+					else nodeChangedSource = NodeChangedSource.History;
```
After changing to ids, we will have to make sure the node is loaded so:
`nodeController.NodeLoadManager.LoadNode(index);
nodeController.OnNodeLoadSessionEnded?.Invoke();`
before selecting

stasgora replied:
Maybe extract those 3 lines to a private function

stasgora commented on Assets/Scripts/Controllers/HistoryController.cs:
```diff
@@ -1,17 +1,55 @@
 		private NetworkController networkController;
 		private GraphController graphController;
 		public HistoryService HistoryService { get; private set; }
+		public static Action startLoading;
```
What is that used for?

stasgora commented on Assets/Scripts/Controllers/HistoryController.cs:
```diff
@@ -21,24 +59,163 @@ public class HistoryController : MonoBehaviour {
+				routesTiles[i] = Instantiate(RouteTemplate, RouteParent.transform);
+				string getFileName = Path.GetFileNameWithoutExtension(name);
+				routesTiles[i].transform.GetChild(0).GetComponent<Text>().text = getFileName;
+				routesTiles[i].transform.GetChild(1).GetComponent<Text>().text = "Route Length: <color=black>" + lengths[i].ToString() + "</color>";
```
this `.ToString()` is redundant

Evil00787 committed:
Changed storing Node to uint?

stasgora committed:
Accept null in network functions

Evil00787 committed:
Shortened selecting

Evil00787 committed:
Merge branch 'feature/WG-109' of https://github.com/wikitools/wikigraph into feature/WG-109

stasgora reviewed (changes requested):
first batch

Evil00787 committed:
Update Assets/Scripts/Controllers/HistoryController.cs

Co-Authored-By: Stanisław Góra <stasgora@gmail.com>

Evil00787 committed:
Resolved suggestions - first batch

Evil00787 committed:
Deleted unused variables

stasgora committed:
Cache node type border

stasgora committed:
Highlighted fix

mikolajmirko committed:
header zero connections exeption

stasgora committed:
Merge remote-tracking branch 'origin/feature/WG-146' into feature/WG-109

mikolajmirko committed:
info space input blocking fix

stasgora committed:
Merge remote-tracking branch 'origin/feature/WG-146' into feature/WG-109

stasgora commented on Assets/Scripts/Services/DataFiles/DataFileReader.cs:
```diff
@@ -59,17 +63,20 @@ public class DataFileReader : IDisposable {
 			streams[type] = new DataFile {Stream = stream, Length = new FileInfo(filePath).Length};
 		}
+		
+		public string GetDataPackFile() => Path.Combine(DATA_FILE_PATH, dataPack, dataPackDate, $"{dataPack}.{DATA_FILE_EXTENSION}");
```
this should call `GetDataPackDirectory` to reduce code dupliaction

stasgora committed:
Rename Line Route to avoid double meaning in code

stasgora commented on Assets/Scripts/Services/History/RoutesFiles/RoutesReader.cs:
```diff
@@ -0,0 +1,79 @@
+	public class RoutesReader : IDisposable {
+		private List<StreamReader> streams = new List<StreamReader>();
+
+		private const string DATA_FILE_EXTENSION = ".wgroute";
```
All our files have 3 letter extensions wg_, maybe we should use wgr here?

stasgora commented on Assets/Scripts/Services/History/RoutesFiles/RoutesReader.cs:
```diff
@@ -0,0 +1,79 @@
+using System.Text;
+using UnityEngine;
+
+namespace Services.RoutesFiles {
```
Put RouteLoader & Reader in Services/Routes directory to match the namespace (change it to Routes for simplicity)

stasgora commented on Assets/Scripts/Services/History/RoutesFiles/RoutesReader.cs:
```diff
@@ -0,0 +1,79 @@
+using UnityEngine;
+
+namespace Services.RoutesFiles {
+	public class RoutesReader : IDisposable {
```
RouteReader? Singular sounds better imo

stasgora commented on Assets/Scripts/Services/History/RoutesFiles/RoutesLoader.cs:
```diff
@@ -0,0 +1,76 @@
+
+
+namespace Services.RoutesFiles {
+	public class RoutesLoader : IDisposable {
```
RouteLoader? Singular sounds better imo

stasgora commented on Assets/Scripts/Services/History/SearchFiles/SearchLoader.cs:
```diff
@@ -0,0 +1,35 @@
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Services.SearchFiles {
```
Same thing, to Services/Search folder + namespace rename for me

stasgora replied:
Maybe both Search & Routes should end up under Console (folder + namespace) but have definitely nothing to do with History

stasgora commented on Assets/Prefabs/ArticleSearchResult.prefab:
```diff
@@ -0,0 +1,293 @@
+%YAML 1.1
```
Put those two prefabs in SearchResults subfolder

stasgora commented on Assets/Scripts/Services/History/RoutesFiles/RoutesLoader.cs:
```diff
@@ -0,0 +1,76 @@
+
+namespace Services.RoutesFiles {
+	public class RoutesLoader : IDisposable {
+		private enum USERACTION_TYPE {
```
Name formatting

Evil00787 replied:
Did you mean USER_ACTION_TYPE?

stasgora replied:
UserActionType, as all other types

stasgora commented on Assets/Scripts/Controllers/HistoryController.cs:
```diff
@@ -21,24 +58,175 @@ public class HistoryController : MonoBehaviour {
+			if (searched != text) {
+				searched = text;
+				if (isSearching) StopCoroutine(searchCoroutine);
+				if (text != string.Empty) {
```
Why not clear the results on empty field?

stasgora commented on Assets/Scripts/Controllers/HistoryController.cs:
```diff
@@ -1,17 +1,54 @@
+using UnityEngine.UI;
 
 namespace Controllers {
 	public class HistoryController : MonoBehaviour {
```
This should ideally be divided into SearchController & RouteController since they are separate things

stasgora commented on Assets/Scripts/Controllers/HistoryController.cs:
```diff
@@ -21,24 +58,175 @@ public class HistoryController : MonoBehaviour {
-					nodeChangedByHistory = false;
+					if (nodeChangedSource != NodeChangedSource.History) {
+						if(oldNode == null) {
+							HistoryService.RegisterAction(new NodeSelectedAction(null, newNode.ID, false));
```
You can simply use the Null-conditional operators (node?.ID)

stasgora commented on Assets/Scripts/Services/History/HistoryService.cs:
```diff
@@ -26,5 +35,79 @@ public class HistoryService {
 		}
+
+		#endregion
+		#region Routes
```
This whole region is basically same as History one but extended. The first region should be extracted as a base class - maybe `ActionService`, and this region as `RouteService: ActionService` without the duplicated functions & fields

stasgora commented on Assets/Scripts/Services/History/HistoryService.cs:
```diff
@@ -26,5 +35,79 @@ public class HistoryService {
+
+		public int[] getLengths() {
+			return routesLoader.routeLengths();
+		}
```
These are just proxies going through Loader to Reader, unnecessary code if you ask me. Remove at least the Loader step if not call directly from the Controller

stasgora commented on Assets/Scripts/Services/History/HistoryService.cs:
```diff
@@ -26,5 +35,79 @@ public class HistoryService {
+
+		public bool isPlayingRoute() {
+			return playsRoute;
+		}
```
Good place for a property

stasgora commented on Assets/Scripts/Services/History/RoutesFiles/RoutesLoader.cs:
```diff
@@ -0,0 +1,76 @@
+		};
+
+		
+
```
All new files could do with a small auto-reformatting ;)

mikolajmirko committed:
header enable function

stasgora commented on Assets/Scripts/Services/History/RoutesFiles/RoutesLoader.cs:
```diff
@@ -0,0 +1,76 @@
+		
+
+		private RoutesReader routesReader;
+		private static char separator = ';';
```
const, caps only

stasgora commented on Assets/Scripts/Services/History/RoutesFiles/RoutesLoader.cs:
```diff
@@ -0,0 +1,76 @@
+
+		private RoutesReader routesReader;
+		private static char separator = ';';
+		public static Func<uint, Node> getRouteNode;
```
This isn't used, correct?

stasgora commented on Assets/Scripts/Services/History/RoutesFiles/RoutesLoader.cs:
```diff
@@ -0,0 +1,76 @@
+
+		public int routesNumber() {
+			return routesReader.numberOfRoutes();
+		}
```
Not used?

stasgora committed:
Rename Data file namespace

stasgora commented on Assets/Scripts/Services/History/RoutesFiles/RoutesReader.cs:
```diff
@@ -0,0 +1,79 @@
+		private string[] fileNames;
+		private int[] routeLength;
+
+		public RoutesReader(string path, string dataFilePostfix = "") {
```
why do we have this `dataFilePostfix` params that start all the way in `HistoryService`? Remove it all if not used

stasgora commented on Assets/Scripts/Services/History/RoutesFiles/RoutesLoader.cs:
```diff
@@ -0,0 +1,76 @@
+			return rev;
+		}
+
+		public void Dispose() {
```
Started implementing the `Dispose` mechanism but it ends here. Should be called from `MonoBehaviours` `OnDestroy`

stasgora commented on Assets/Scripts/Services/History/SearchFiles/SearchLoader.cs:
```diff
@@ -0,0 +1,35 @@
+		public SearchReader reader;
+		
+
+		public SearchLoader(int n, string path) {
```
`n`?

stasgora reviewed (changes requested):
More work to do

mikolajmirko committed:
AUTO icon binding

mikolajmirko committed:
Merge branch 'feature/WG-109' of https://github.com/wikitools/wikigraph into feature/WG-109

stasgora committed:
Merge remote-tracking branch 'origin/feature/WG-146' into feature/WG-109

stasgora committed:
Header update

stasgora committed:
Route play state syncing

stasgora commented on Assets/Scripts/Controllers/HistoryController.cs:
```diff
@@ -1,44 +1,235 @@
+		public GameObject SearchScrollView;
+		
+		public static Action startRouteAutoAction;
+		public static Action endRouteAutoAction;
```
- why are they static?
- usual naming conversion would be `OnRouteStarted/Ended`
- should end up in `(Route)Controller` so they can be easily reached from other parts of application (actually had to move them myself when adding syncing for the AUTO icon)

stasgora replied:
After seeing the action usage in Header it would be better to replace these two events with just one: `OnRoutePlayStateChanged<bool>` since the usage is mirrored anyway

mikolajmirko committed:
header indicator and auto icon toggle fix

stasgora commented on Assets/Scripts/Services/History/HistoryService.cs:
```diff
@@ -26,5 +36,79 @@ public class HistoryService {
+				if (redoSavedRoute.Count > 0) yield return new UnityEngine.WaitForSeconds(secondsToNextRoute);
+			}
+			playsRoute = false;
+			controller.NetworkController.SyncRoutePlaying(false);
```
Since those events need to go through a sync before being distributed as `Action` events (on clients) there should be a single point of entry for these calls

stasgora replied:
The Route-stop-playing signal is sent only when the route ends, not when the user does something that cancels it. Need to handle all those situations similarly.

Evil00787 committed:
Fixes to pull request

Evil00787 committed:
AUTO icon now works properly
More fixes

Evil00787 committed:
Major refactoring - new Controller:
HistoryController has been divided to:
	-ActionController
	-RouteController
	-SearchController
Similar division of HistoryService (without unnecesary SearchService)

Evil00787 committed:
It works! Pull request changes

Evil00787 committed:
lzwp scene updated

Evil00787 committed:
One more lzwp scene change

Evil00787 committed:
Fixed bug with needed double click to play the same foute after being interrupted by different source

Evil00787 committed:
Fixed routes syncing

Evil00787 committed:
small changes

mikolajmirko committed:
connection count 0 fix (i hope it wont break sth else)

stasgora committed:
Merge branch 'master' into feature/WG-109

stasgora committed:
Read 3 bytes of parent count

mikolajmirko committed:
indicator min width

stasgora committed:
Fixing races in loaded node client syncing

mikolajmirko committed:
null exceptions and indicator not visible on client fix

Evil00787 committed:
No more NullPointerExceptions on console close!
Made Focus in searchInput great again!
Scrolling connections now stop autoRoute! :O

Evil00787 committed:
Modified gitignore

stasgora committed:
Bring back the test pack

stasgora committed:
Distribution fallback

stasgora committed:
Scene UI groups

stasgora committed:
+ grid
